core/src/impl/Kokkos_SharedAlloc.cpp

```diff
@@ -285,48 +285,19 @@ void SharedAllocationRecord<void, void>::print_host_accessible_records(
   // allocation.
   const SharedAllocationRecord<void, void>* r = root->m_next;
 
-  char buffer[256];
-
   if (detail) {
     while (r != root) {
-      // Formatting dependent on sizeof(uintptr_t)
-      const char* format_string;
-
-      if (sizeof(uintptr_t) == sizeof(unsigned long)) {
-        format_string =
-            "%s addr( 0x%.12lx ) list( 0x%.12lx 0x%.12lx ) extent[ 0x%.12lx + "
-            "%.8ld ] count(%d) dealloc(0x%.12lx) %s\n";
-      } else if (sizeof(uintptr_t) == sizeof(unsigned long long)) {
-        format_string =
-            "%s addr( 0x%.12llx ) list( 0x%.12llx 0x%.12llx ) extent[ "
-            "0x%.12llx + %.8ld ] count(%d) dealloc(0x%.12llx) %s\n";
-      }
-
-      snprintf(buffer, 256, format_string, space_name,
-               reinterpret_cast<uintptr_t>(r),
-               reinterpret_cast<uintptr_t>(r->m_prev),
-               reinterpret_cast<uintptr_t>(r->m_next),
-               reinterpret_cast<uintptr_t>(r->m_alloc_ptr), r->m_alloc_size,
-               r->use_count(), reinterpret_cast<uintptr_t>(r->m_dealloc),
-               r->m_alloc_ptr->m_label);
-      s << buffer;
+      s << space_name << " addr( " << r << " ) list ( " << r->m_prev << ' '
+        << r->m_next << " ) extent[ " << r->m_alloc_ptr << " + "
+        << r->m_alloc_size << " ] count(" << r->use_count() << ") dealloc("
+        << reinterpret_cast<void*>(r->m_dealloc) << ") "
+        << r->m_alloc_ptr->m_label << '\n';
```

Comment:
Please comment about the formatting.  Did you compare before and after?
Any reason why you drop the precision?  You could have used `std::setw(12)` or whatnot.

Reply:
See https://github.com/kokkos/kokkos/pull/6354#issuecomment-1681347629 to me that looked close enough.

Reply:
Please have another look. We now have
```
Host addr( 0x600002be51f0 ) list( 0x0001009fa1c0 0x600002be5180 ) extent[ 0x0001495f7840 + 00009728 ] count(120) dealloc(0x000100939090) test_1199
Host addr( 0x600002be5180 ) list( 0x600002be51f0 0x600002be5110 ) extent[ 0x0001495f5040 + 00009720 ] count(120) dealloc(0x000100939090) test_1198
Host addr( 0x600002be5110 ) list( 0x600002be5180 0x600002be50a0 ) extent[ 0x0001495f2840 + 00009712 ] count(120) dealloc(0x000100939090) test_1197
Host addr( 0x600002be50a0 ) list( 0x600002be5110 0x600002be5030 ) extent[ 0x0001495f0040 + 00009704 ] count(120) dealloc(0x000100939090) test_1196
```
before and
```
Host addr( 0x600001e4c3c0 ) list ( 0x00010294dfb0 0x600001e4c360 ) extent[ 0x0001465f6a40 + 00009728 ] count(120) dealloc(0x0001028f026c) test_1199
Host addr( 0x600001e4c360 ) list ( 0x600001e4c3c0 0x600001e4c300 ) extent[ 0x0001465f4240 + 00009720 ] count(120) dealloc(0x0001028f026c) test_1198
Host addr( 0x600001e4c300 ) list ( 0x600001e4c360 0x600001e4c2a0 ) extent[ 0x0001465f1a40 + 00009712 ] count(120) dealloc(0x0001028f026c) test_1197
Host addr( 0x600001e4c2a0 ) list ( 0x600001e4c300 0x600001e4c240 ) extent[ 0x0001465ef240 + 00009704 ] count(120) dealloc(0x0001028f026c) test_1196
```
after